### PR TITLE
Tilting at contours of windmills

### PR DESCRIPTION
**Pull request by rsheeter**
Maybe this is reflective of being a noob more than anything else but the combination of the nested function and std::iter::from_fn feels hard to follow. I tried to make it more idiomatic.

Better? - if so please approve.
Worse? - if so please explain why so I can learn.

### CODE REVIEW

File: write-fonts/src/tables/glyf.rs [hunk 1]
```diff
@@ -202,62 +202,90 @@ impl SimpleGlyph {
...
     pub fn contours(&self) -> &[Contour] {
         &self.contours
     }
 }
 
+// this is adapted from simon's implementation at
+// https://github.com/simoncozens/rust-font-tools/blob/105436d3a617ddbebd25f790b041ff506bd90d44/fonttools-rs/src/tables/glyf/glyph.rs#L268
+struct PointDeltaIter<'a, I>
+where
+    I: Iterator<Item = &'a CurvePoint>,
```
Comment (dfrg) [inline write-fonts/src/tables/glyf.rs:220]: The bound on `I` here should be necessary.

File: write-fonts/src/tables/glyf.rs [hunk 2]
```diff
@@ -202,62 +202,90 @@ impl SimpleGlyph {
...
+{
+    last_x: i16,
+    last_y: i16,
+    points: I,
+}
+
+impl<'a, I> PointDeltaIter<'a, I>
+where
+    I: Iterator<Item = &'a CurvePoint>,
```
Comment (dfrg) [inline write-fonts/src/tables/glyf.rs:229]: ditto

**Review by dfrg — APPROVED** [review]
Approved with some small suggestions. I don’t have a strong feeling on this one way or the other so merge at your discretion.

File: write-fonts/src/tables/glyf.rs (hunk 1 above)
Comment (rsheeter) [reply write-fonts/src/tables/glyf.rs:220]: Huh. I'd have sworn it complained when I didn't put that.

**Comment by rsheeter** [comment]
I don't think I'm convinced this is clearly better. It's certainly the way I personally prefer it, with less nesting and lots of small separate bits, but that doesn't seem enough to justify actually changing it.
